core/modes/diff_mode.go

```diff
@@ -124,9 +126,24 @@ func nullOrValue(value interface{}) string {
 	return fmt.Sprint(value)
 }
 
-func (this *DiffMode) headerDiff(expected map[string][]string, actual map[string][]string) bool {
+func (this *DiffMode) headerDiff(expected map[string][]string, actual map[string][]string, headersWhitelist []string) bool {
 	same := true
 	for k := range expected {
+		// if len(headersWhitelist) == 1 && headersWhitelist[0] != "*" {
+		// 	continue
+		// }
+		shouldContinue := true
+		if len(headersWhitelist) > 0 {
+			shouldContinue = false
+		}
+		for _, header := range headersWhitelist {
+			if k == header || header == "*" {
+				shouldContinue = true
+			}
+		}
+		if !shouldContinue {
```

Comment:
The check suggests `should not continue` but it does `continue` to go to the next iteration. Maybe `shouldDoDiff`or `isWhitelistedForDiffing` would be a better name?